### DISCUSSION TITLE
Directive confirm that pre-requisites is installed on the box.

### DISCUSSION THREAD

**Discussion by larssb** [PowerShellOrg/Plaster#392] [Ideas]
This is a follow up to comments in #5 from 170623.

Useful if the project being generated with Plaster is a NodeJS project, uses TypeScript or something down that lane. In such cases it wold be great to be able to control that the user has NodeJS installed and so forth.

Thank you.

**Comment by rkeithhill** [comment]
How about I add a new built-in variable called `$PLASTER_AvailableCommands` that I scoop up with `Get-Command -CommandType All`.  You can then use it like so
```xml
<file condition='$PLASTER_AvailableCommands.Where({$_.Name -eq "node.exe" -and $_.CommandType -eq "Application"}).Version.Major -ge 6' ...
```

**Comment by larssb** [comment]
A totally legit idea.

On MacOS v6 beta3 of PowerShell core I don't get a version number in the output. See the attached picture. I have no idea why, unfortunately. Tried with a different set of executables. Same result.
Checked for a newer version of powershell for linux/mac. There is none per 170702.

<img width="346" alt="screen shot 2017-07-02 at 11 08 11" src="https://user-images.githubusercontent.com/5261530/27768567-0172217a-5f17-11e7-8557-42351c0f3f20.png">

The command I ran was `Get-command -Name git -CommandType All`

**Comment by rkeithhill** [comment]
Accessing the version info will only be valid on Windows as Windows binaries support encoding the version number as a resource in the binary.   Either ELF binaries don't do support this or PS / .NET Core don't support reading the version info yet.  I suspect the former.

**Comment by larssb** [comment]
@rkeithhill dang, that explains it. Hmm, maybe @daviwil knows a thing or two about this on MacOS, if I may be so free to pull you into the conversation here :-). There must be some other way on Linux/MacOS

Thank you.

**Comment by larssb** [comment]
Any update on this issue?
